### PR TITLE
send settings to all iframes

### LINKED ISSUE #2041: ipcRenderer events are not received inside iframes
messages sent by `this.webview.send` are not received by `brave-default.js` running inside an iframe. as a result, ads are not replaced in https://zyan.scripts.mit.edu/brave/slashdot.html and fingerprinting protection will not always work inside iframes.

i would expect the top-level instance of `brave-default.js` to forward the messages onto subframes.

blocks #313 somewhat, but can be worked around

cc @bridiver

### PR DESCRIPTION
**Pull request by bridiver**
- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Ran `git rebase -i` to squash commits if needed.

requires https://github.com/brave/electron/commit/42ffb5c19830ea73f29eb9a32d78a9a527d81f1e
fixes #2041 

@bbondy @diracdeltas

### CODE REVIEW

**Comment by bridiver** [comment]
this currently crashes with private tabs

**Comment by bridiver** [comment]
fixed with https://github.com/brave/electron/commit/0b58df348733014baa3b955dcdddca1f5a1ad732

File: app/extensions.js [hunk 1]
```diff
@@ -15,23 +17,51 @@ let generateBraveManifest = () => {
       {
         run_at: 'document_start',
         all_frames: true,
-        matches: ['<all_urls>'],
+        matches: ['http://*/*', 'https://*/*', 'file://*'],
```
Comment (diracdeltas) [inline app/extensions.js]: also `data:` and probably `blob:`?

Comment (diracdeltas) [inline app/extensions.js]: pretty sure they are needed at least for `data:` URLs because those can be HTML documents

Comment (bridiver) [inline app/extensions.js]: ok, should be pretty easy change

**Comment by diracdeltas** [comment]
this looks much cleaner, thanks.

**Comment by bridiver** [comment]
I think there is still a lot we can do to clean things up using the new background page functionality, but I didn't want to do too much in one PR

**Comment by bbondy** [comment]
Is this ready to go? I think you can merge after fixing rebase.

**Comment by bridiver** [comment]
it requires an updated electron build, but I do want to get this merged asap

**Comment by diracdeltas** [comment]
i'm fine with that. lmk when upstream electron build is available over npm.

**Comment by bbondy** [comment]
We're at 1.2.2 now, merging.

**Pull request merged**